test/unit/sites.js

```diff
@@ -3,179 +3,325 @@
+        done();
+      });
+    });
+  });
+
+  describe('createSite(req)', function() {
+    it('create a site', function(done) {
       var newData = {
-        site_id: 77777,
         site_name: 'newSiteName',
         site_address: 'new Boston, MA'
       };
-      sites.getSites()
+      sites.getSites({})
       .then(function(data) {
-        assert.deepEqual([fakeData1, fakeData2], data);
+        assert.deepEqual([site1, site2, site3, site4, site5, site6, site7, site8, site9, site10, site11], data);
 
         return sites.postSite({
-          data: {
-            site: {
-              site_id: newData.site_id,
-              site_name: newData.site_name,
-              site_address: newData.site_address
-            }
+          body: {
+            site_name: newData.site_name,
+            site_address: newData.site_address
           }
         });
       })
       .then(function() {
         return sites.getSites({});
       })
       .then(function(data) {
-        assert.deepEqual([fakeData1, fakeData2, newData], data);
+        assert.deepEqual([site1, site2, site3, site4, site5, site6, site7, site8, site9, site10, site11, {
+          site_id: 12,
+          site_name: newData.site_name,
+          site_address: newData.site_address
+        }], data);
+        done();
+      });
+    });
+
+    it('does not create a site because body was missing', function(done) {
+      sites.postSite({}).then(function() {
+        return sites.getSites({});
+      })
+      .catch(function(err) {
+        assert.equal(err.status, 406);
+        assert.equal(err.message, 'Must provide site\'s name, and address');
+        done();
+      });
+    });
+
+    it('does not create a site because the site_name was missing', function(done) {
+      sites.postSite({
+        body: {
+          site_address: 'address'
+        }
+      })
+      .then(function() {
+        return sites.getSites({});
+      })
+      .catch(function(err) {
+        assert.equal(err.status, 406);
+        assert.equal(err.message, 'Must provide site\'s name, and address');
+        done();
+      });
+    });
+
+    it('does not create a site because site_address was missing', function(done) {
+      sites.postSite({
+        body: {
+          site_name: 'name'
+        }
+      })
+      .then(function() {
+        return sites.getSites({});
+      })
+      .catch(function(err) {
+        assert.equal(err.status, 406);
+        assert.equal(err.message, 'Must provide site\'s name, and address');
+        done();
+      });
+    });
+  });
+
+  describe('getSite(req)', function() {
+    it('get a site', function(done) {
+      var promise = sites.getSite({
+        params: {
+          site_id: 1
+        }
+      });
+
+      promise.then(function(data) {
+        assert.deepEqual([site1], data);
         done();
       });
     });
 
-    xit('get a site', function(done) {
+    it('gets a site that does not exist', function(done) {
```

Comment:
Again, probably clearer if it says "Returns an empty array if the site_id doesn't exist"